### PR TITLE
Adds load balancer CIDRs and Storage Account Type

### PR DESCRIPTION
**Pull request by rossf7**
Towards #6 

This PR adds config for deploying the main template.

* Storage Account Types -  we use `Standard_LRS` but there is also a premium tier so we make this configurable
* Load Balancer CIDRs - for IPAM we need to assign the CIDRs for the Virtual Network. For Azure this includes the 3 load balancers we create for etcd, apiserver and ingress

### CODE REVIEW

File: spec.go [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package azuretpr
 
 import (
-	"github.com/giantswarm/azuretpr/spec"
 	"github.com/giantswarm/clustertpr"
+
+	"github.com/giantswarm/azuretpr/spec"
```
Comment (xh3b4sd) [inline spec.go:6]: <3

File: spec/azure/storage.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package azure
+
+type Storage struct {
+	AccountType string `json:"accountType" yaml:"accountType"`
```
Comment (xh3b4sd) [inline spec/azure/storage.go]: Lets have the comments you provided with the PR description also here in the code. This is more sustainable and helpful for everybody coming by.

File: spec/azure/virtual_network.go [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
 type VirtualNetwork struct {
-	CIDR             string `json:"cidr" yaml:"cidr"`
-	MasterSubnetCIDR string `json:"masterSubnetCidr" yaml:"masterSubnetCidr"`
-	WorkerSubnetCIDR string `json:"workerSubnetCidr" yaml:"workerSubnetCidr"`
+	CIDR                    string `json:"cidr" yaml:"cidr"`
+	MasterSubnetCIDR        string `json:"masterSubnetCidr" yaml:"masterSubnetCidr"`
+	WorkerSubnetCIDR        string `json:"workerSubnetCidr" yaml:"workerSubnetCidr"`
+	APILoadBalancerCIDR     string `json:"apiLoadBalancerCidr" yaml:"apiLoadBalancerCidr"`
+	EtcdLoadBalancerCIDR    string `json:"etcdLoadBalancerCidr" yaml:"etcdLoadBalancerCidr"`
+	IngressLoadBalancerCIDR string `json:"ingressLoadBalancerCidr" yaml:"ingressLoadBalancerCidr"`
```
Comment (xh3b4sd) [inline spec/azure/virtual_network.go]: From the user point of view I would like to do this. 

```
customObject.Spec.VitualNetwork.CIDR.Guest.Master
customObject.Spec.VitualNetwork.CIDR.Guest.Worker
customObject.Spec.VitualNetwork.CIDR.LB.API
customObject.Spec.VitualNetwork.CIDR.LB.Etcd
customObject.Spec.VitualNetwork.CIDR.LB.Ingress
```

I have no string feelings about the specific scheme and block names here. I just want to say that a more structured structure is more readable and easier to use for the actual user.

File: spec/azure/storage.go (hunk 2 above)
Comment (rossf7) [reply spec/azure/storage.go]: Done :)

File: spec/azure/virtual_network.go (hunk 3 above)
Comment (rossf7) [reply spec/azure/virtual_network.go]: Good point. I've restructured the LoadBalancer CIDRs to a separate block which I think makes it cleaner.

**Comment by rossf7** [comment]
@xh3b4sd PTAL

**Review by xh3b4sd — APPROVED** [review]
I personally would have structured the blocks a bit more. When you think this is right lets go.

**Comment by rossf7** [comment]
Ack. Going to go with this for now as I think having CIDR in the field names makes sense. There is still a lot of Azure config to be added.  So happy to refactor later if we need to structure more.

File: spec/azure/loadbalancer/loadbalancer.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package loadbalancer
```
Comment (kopiczko) [inline spec/azure/loadbalancer/loadbalancer.go:1]: This should sit in virtualnetwork package

Comment (rossf7) [reply spec/azure/loadbalancer/loadbalancer.go:1]: Ack, fix is in https://github.com/giantswarm/azuretpr/pull/4

**Pull request merged**